### PR TITLE
feat(logging): setup server logging

### PR DESCRIPTION
**Pull request by bassrock**
## Goal

- Move list-api to the new logging pattern
- Update to Node 18.16
- Remove github linter for node linter

## I'd love feedback/perspectives on:
- 

## Implementation Decisions
- 

## Deployment steps
- [ ] Database migrations?
- [ ] Secrets?
- [ ] Post in [\#changelog](https://pocket.slack.com/archives/C0Q4UFMDZ):
> Write a #changelog message using our [best practices](https://docs.google.com/document/d/1oEt8Mtkp-6Xz9S2zaNX1EIXfQIEDN2JpY0diuPc1HZc/edit) if this PR (potentially) impacts users. Describe the 'why' and 'what'. @mention relevant teams. Link to this PR.

## References

Jira ticket:
* https://getpocket.atlassian.net/browse/BACK-

Documentation:
* Project doc

### CODE REVIEW

File: src/resolvers/utils.ts [hunk 1]
```diff
@@ -74,7 +74,7 @@ export function convertToTagSaveAssociations(
  * @param attr The attribute of the parent to request
  * @returns The value keyed by `attr` of the parent or fetched object.
  */
-export async function lazyParentLoad<T, K, A extends keyof T>(
+export async function lazyParentLoad<T extends object, K, A extends keyof T>(
```
Comment (bassrock) [inline src/resolvers/utils.ts:77]: I did this change based on some quick internet research. Looks like the latest typescript required this.

File: .aws/src/main.ts [hunk 2]
```diff
@@ -311,6 +312,8 @@ class ListAPI extends TerraformStack {
               valueFrom: databaseSecretEnvVars.writePassword,
             },
           ],
+          logGroup: this.createCustomLogGroup('app'),
+          logMultilinePattern: '^\\S.+',
```
Comment (jpetto) [inline .aws/src/main.ts:316]: this never had logs? 😬

File: .circleci/config.yml [hunk 3]
```diff
@@ -332,6 +338,8 @@ workflows:
           sentry_env: production
           sentry_org: pocket
           workspace-path: /tmp/workspace
+          source-maps-path: /tmp/workspace/app_prod/dist
+          inject-source-maps: true
```
Comment (jpetto) [inline .circleci/config.yml:342]: 😅

File: src/businessEvents/unifiedEventKinesisHandler.ts [hunk 4]
```diff
@@ -109,12 +110,17 @@ export async function unifiedEventKinesisHandler(
...
-    } event(s) to kinesis stream '${
-      config.aws.kinesis.unifiedEvents.streamName
-    }'. Failed Events: \n ${JSON.stringify(failedEvents)}`;
-    console.log(errorMessage);
-    Sentry.captureException(new Error(errorMessage));
+    serverLogger.error('Failed to send event(s) to kinesis stream', {
+      stream: config.aws.kinesis.unifiedEvents.streamName,
+      events: JSON.stringify(failedEvents),
+    });
+    Sentry.addBreadcrumb({
```
Comment (jpetto) [inline src/businessEvents/unifiedEventKinesisHandler.ts:117]: maybe a tangent - did we figure out @efixler's questions about following breadcrumbs? just wondering how useful this is with `ts-logger` in place.

File: src/aws/eventBridgeBase.ts [hunk 5]
```diff
@@ -27,18 +28,18 @@ export class EventBridgeBase {
     try {
       const output: PutEventsCommandOutput = await this.client.send(command);
       if (output.FailedEntryCount) {
+        serverLogger.error(failedEventError);
```
Comment (jpetto) [inline src/aws/eventBridgeBase.ts:31]: definitely not a blocker, but what do you think about adding more structure to these calls as a general practice? something like `serverLogger.error('failed putting events to event bridge', { failedEventError, originalError: error});`

**Review by jpetto — APPROVED** [review]
a couple questions but no blockers. ty for adding this in! (did _not_ know logging was missing in this service 😅.)

File: .circleci/config.yml (hunk 3 above)
Comment (bassrock) [reply .circleci/config.yml:342]: Lol right.

File: src/businessEvents/unifiedEventKinesisHandler.ts (hunk 4 above)
Comment (bassrock) [reply src/businessEvents/unifiedEventKinesisHandler.ts:117]: Yea. I think long term, ts-logger needs to emit to Sentry breadcrumbs, but we just haven't set a pattern on making that happen.

File: src/aws/eventBridgeBase.ts (hunk 5 above)
Comment (bassrock) [reply src/aws/eventBridgeBase.ts:31]: I agree, could we follow up with that type of stuff?

Comment (jpetto) [reply src/aws/eventBridgeBase.ts:31]: yeah, 100%. no reason to block this PR on that.

File: .aws/package.json [hunk 6]
```diff
@@ -18,10 +18,10 @@
     "node": ">=12"
   },
   "dependencies": {
-    "@pocket-tools/terraform-modules": "4.15.0"
+    "@pocket-tools/terraform-modules": "4.17.0"
```
Comment (bassrock) [inline .aws/package.json:21]: Not going to the latest because it requires state manipulation to migrate.

**Pull request merged**